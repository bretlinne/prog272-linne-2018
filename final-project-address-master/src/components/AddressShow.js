import React, {Component} from 'react';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import AddressEdit from './AddressEdit';
import blue from '@material-ui/core/colors/blue';
import MuiButton from '@material-ui/core/Button';
import Icon from '@material-ui/core/Icon';

const styles = theme => ({
    rightIcon: {
        marginLeft: theme.spacing.unit,
    },
    button: {
        margin: theme.spacing.unit
    },
    container: {
        flexGrow: 1,
        textAlign: 'center'
    },
    rootBar: theme.mixins.gutters({
        paddingTop: 16,
        paddingBottom: 16,
        paddingLeft: 8,
        paddingRight: 8,
        marginTop: theme.spacing.unit * 3,
        marginLeft: theme.spacing.unit * 3,
        marginRight: theme.spacing.unit * 3
    }),
});

class AddressShow extends Component {
    constructor(props) {
        super(props);
        this.state = {
            editOpen: false,
            edits: {
                _id: 'unknown',
                firstName: 'unknown',
                lastName: 'unknown',
                address: 'unknown',
                city: 'unknown',
                state: 'unknown',
                zip: 'unknown',
                phone: 'unknown',
                contact: 'unknown'
            }
        };
    }
    setEdits = (name, event) => {
        const data = this.props.name;
        //var data = this.props.name;
        data[name] = event.target.value;
        this.setState({edits: data});
    };

    addressEdit = address => {
        console.log(address);

        if (!address) {
            return this.setState({editOpen: false});
        }

        this.setState({
            edits: address,
            editOpen: false
        });
    };

    save = () => {
        this.props.save(this.state.edits);
    };

    render() {
        const {classes} = this.props;
        const editDialog = this.state.editOpen ? (
            <AddressEdit
                address={this.props.name}
                open={this.state.editOpen}
                addressEdit={this.addressEdit}
            />
        ) : (
            <div/>
        );

        return (
            <div className={classes.container}>
                <Paper className={classes.rootBar}>
                    <div className="App">
                        <MuiButton
                            color={blue}
                            variant='raised'
                            onClick={(e) => this.props.setAddress(-10, e)}
                        >
                            <Icon>fast_rewind</Icon>
                        </MuiButton>
                        <MuiButton
                            color={blue}
                            variant='raised'
                            onClick={(e) => this.props.setAddress(-1, e)}
                        >
                            <Icon>arrow_back_ios</Icon>
                        </MuiButton>
                        <MuiButton
                            color={blue}
                            variant='raised'
                            onClick={this.props.showAddress}
                        >
                            <Icon>cloud_download</Icon>
                        </MuiButton>
                        <MuiButton
                            color={blue}
                            variant='raised'
                            onClick={event => this.props.setAddress(1, event)}
                        >
                            <Icon>arrow_forward_ios</Icon>
                        </MuiButton>
                        <MuiButton
                            color={blue}
                            variant='raised'
                            onClick={event => this.props.setAddress(10, event)}
                        >
                            <Icon>fast_forward</Icon>
                        </MuiButton>

                    </div>
                    <div className="App">
                        <MuiButton
                            color={blue}
                            variant='raised'
                            onClick={this.save}
                        >
                            <Icon>save</Icon>
                        </MuiButton>
                        <MuiButton
                            color={blue}
                            variant='raised'
                            onClick={event =>
                                this.props.delete(this.props.name, event)
                            }
                        >
                            <Icon>delete</Icon>
                        </MuiButton>
                        <MuiButton
                            color={blue}
                            variant='raised'
                            onClick={() => this.setState({editOpen: true})}
                        >
                            <Icon>edit</Icon>
                        </MuiButton>
                        {editDialog}
                    </div>
                    <p className="App-intro">
                        First Name: {this.props.name.firstName}
                    </p>
                    <p className="App-intro">
                        Last Name: {this.props.name.lastName}
                    </p>
                    <p className="App-intro">
                        Address: {this.props.name.address}
                    </p>
                    <p className="App-intro">City: {this.props.name.city}</p>
                    <p className="App-intro">State: {this.props.name.state}</p>
                    <p className="App-intro">Zip: {this.props.name.zip}</p>
                    <p className="App-intro">Phone: {this.props.name.phone}</p>
                    <p className="App-intro">Online: {this.props.name.contact}</p>

                </Paper>
            </div>
        );
    }
}
AddressShow.propTypes = {
    classes: PropTypes.object.isRequired
};
export default withStyles(styles)(AddressShow);

