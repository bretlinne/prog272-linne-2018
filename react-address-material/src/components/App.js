import React, { Component } from 'react';
import '../App.css';
import Address from './Address';
import {BrowserRouter, Route} from 'react-router-dom';
import Header from './Header';
import GetFile from './GetFile';

class App extends Component {
  render() {
    return (
        <BrowserRouter>
          <div className="App">
              <Header></Header>
              <Route exact path="/" component={Address}/>
              <Route path="/get-file" component={GetFile}/>
          </div>
        </BrowserRouter>
    );
  }
}

export default App;
