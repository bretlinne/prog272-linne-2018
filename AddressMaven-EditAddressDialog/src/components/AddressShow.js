import React, { Component } from 'react';
import '../App.css';
import '../address-list';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import Icon from '@material-ui/core/Icon';
import styles from './styles';
import AddressEdit from './AddressEdit';
import { withStyles } from '@material-ui/core/styles';

class AddressShow extends Component {
    constructor(props) {
        super(props);
        this.state = {
            editOpen: false,
            edits: {
                _id: 'unknown',
                firstName: 'unknown',
                lastName: 'unknown',
                address: 'unknown',
                city: 'unknown',
                state: 'unknown',
                zip: 'unknown',
                phone: 'unknown',
                contact: 'unknown'
            }
        };
        this.debug = false;
        if (this.debug) {
            console.log('ADDRESS-SHOW CONSTRUCTOR', this.props.address);
        }
    }
    setEdits = (name, event) => {
        const data = this.props.name;
        //var data = this.props.name;
        data[name] = event.target.value;
        this.setState({edits: data});
    };

    addressEdit = address => {
        console.log(address);

        if (!address) {
            return this.setState({editOpen: false});
        }

        this.setState({
            edits: address,
            editOpen: false
        });
    };

    save = () => {
        this.props.save(this.state.edits);
    };
    log(message, message2 = '', message3 = '') {
        if (this.debug) {
            console.log(message, message2, message3);
        }
    }

    render() {
        if (this.debug) {
            console.log('ADDRESS-SHOW RENDER', this.props);
        }
        const editDialog = this.state.editOpen ? (
            <AddressEdit
                address={this.props.address}
                open={this.state.editOpen}
                addressEdit={this.addressEdit}
            />
        ) : (
            <div/>
        );
        return (
            <div className="App">
                <Button
                    id={"leftTen"}
                    color='primary'
                    style={styles.button}
                    variant='raised'
                    onClick={(e) => this.props.setAddress(-10, e)}>
                    <Icon>fast_rewind</Icon>
                </Button>
                <Button
                    id={"leftOne"}
                    color='primary'
                    style={styles.button}
                    variant='raised'
                    onClick={(e) => this.props.setAddress(-1, e)}>
                    <Icon>arrow_back_ios</Icon>
                </Button>
                <Button
                    id={"fetchAddresses"}
                    color='primary'
                    style={styles.button}
                    variant='raised'
                    onClick={(e) => this.props.setAddress(0, e)}>
                    <Icon>cloud_download</Icon>
                </Button>
                <Button
                    id={"rightOne"}
                    color='primary'
                    style={styles.button}
                    variant='raised'
                    onClick={(e) => this.props.setAddress(1, e)}>
                    <Icon>arrow_forward_ios</Icon>
                </Button>
                <Button
                    id={"rightTen"}
                    color='primary'
                    style={styles.button}
                    variant='raised'
                    onClick={(e) => this.props.setAddress(10, e)}>
                    <Icon>fast_forward</Icon>
                </Button>
                <div>
                    <Button
                        id={"edit"}
                        color='secondary'
                        style={styles.button}
                        variant='raised'
                        onClick={() => this.setState({editOpen: true})}
                    >
                        <Icon>edit</Icon>
                    </Button>
                    {editDialog}
                </div>

                <p className="App-intro">
                    First Name: {this.props.address.firstName}
                </p>
                <p className="App-intro">
                    Last Name: {this.props.address.lastName}
                </p>
                <p className="App-intro">
                    Address: {this.props.address.address}
                </p>
                <p className="App-intro">City: {this.props.address.city}</p>
                <p className="App-intro">State: {this.props.address.state}</p>
                <p className="App-intro">Zip: {this.props.address.zip}</p>
                <p className="App-intro">Phone: {this.props.address.phone}</p>
                <p className="App-intro">Fax: {this.props.address.fax}</p>
                <p className="App-intro">
                    Toll Free: {this.props.address.tollfree}
                </p>
            </div>
        );
    }
}
AddressShow.propTypes = {
    classes: PropTypes.object.isRequired
    /*address: PropTypes.shape({
        firstName: PropTypes.string,
        lastName: PropTypes.string,
        address: PropTypes.string,
        city: PropTypes.string,
        state: PropTypes.string,
        zip: PropTypes.string,
        phone: PropTypes.string,
        contact: PropTypes.string
    }),
    setAddress: PropTypes.func,
    fetchAddress: PropTypes.func*/
};
export default withStyles(styles)(AddressShow);
