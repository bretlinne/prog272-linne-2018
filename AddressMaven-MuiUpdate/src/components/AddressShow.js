import React, { Component } from 'react';
import '../App.css';
import '../address-list';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import Icon from '@material-ui/core/Icon';
import styles from './styles';

class AddressShow extends Component {
    constructor(props) {
        super(props);

        this.debug = false;
        if (this.debug) {
            console.log('ADDRESS-SHOW CONSTRUCTOR', this.props.address);
        }
    }

    log(message, message2 = '', message3 = '') {
        if (this.debug) {
            console.log(message, message2, message3);
        }
    }

    render() {
        if (this.debug) {
            console.log('ADDRESS-SHOW RENDER', this.props);
        }
        return (
            <div className="App">
                <Button
                    id={"leftTen"}
                    color='primary'
                    style={styles.button}
                    variant='raised'
                    onClick={(e) => this.props.setAddress(-10, e)}>
                    <Icon>fast_rewind</Icon>
                </Button>
                <Button
                    id={"leftOne"}
                    color='primary'
                    style={styles.button}
                    variant='raised'
                    onClick={(e) => this.props.setAddress(-1, e)}>
                    <Icon>arrow_back_ios</Icon>
                </Button>
                <Button
                    id={"fetchAddresses"}
                    color='primary'
                    style={styles.button}
                    variant='raised'
                    onClick={(e) => this.props.setAddress(0, e)}>
                    <Icon>cloud_download</Icon>
                </Button>
                <Button
                    id={"rightOne"}
                    color='primary'
                    style={styles.button}
                    variant='raised'
                    onClick={(e) => this.props.setAddress(1, e)}>
                    <Icon>arrow_forward_ios</Icon>
                </Button>
                <Button
                    id={"rightTen"}
                    color='primary'
                    style={styles.button}
                    variant='raised'
                    onClick={(e) => this.props.setAddress(10, e)}>
                    <Icon>fast_forward</Icon>
                </Button>

                <p className="App-intro">
                    First Name: {this.props.address.firstName}
                </p>
                <p className="App-intro">
                    Last Name: {this.props.address.lastName}
                </p>
                <p className="App-intro">
                    Address: {this.props.address.address}
                </p>
                <p className="App-intro">City: {this.props.address.city}</p>
                <p className="App-intro">State: {this.props.address.state}</p>
                <p className="App-intro">Zip: {this.props.address.zip}</p>
                <p className="App-intro">Phone: {this.props.address.phone}</p>
                <p className="App-intro">Fax: {this.props.address.fax}</p>
                <p className="App-intro">
                    Toll Free: {this.props.address.tollfree}
                </p>
            </div>
        );
    }
}
AddressShow.propTypes = {
    address: PropTypes.shape({
        firstName: PropTypes.string,
        lastName: PropTypes.string,
        address: PropTypes.string,
        city: PropTypes.string,
        state: PropTypes.string,
        zip: PropTypes.string,
        phone: PropTypes.string,
        fax: PropTypes.string,
        tollfree: PropTypes.string
    }),
    setAddress: PropTypes.func,
    fetchAddress: PropTypes.func
};
export default AddressShow;
