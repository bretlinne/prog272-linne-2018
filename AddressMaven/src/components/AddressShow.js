import React, { Component } from 'react';
import '../App.css';
import '../address-list';
import PropTypes from 'prop-types';
// Put this code near the top
import RaisedButton from 'material-ui/RaisedButton';
//import ActionAndroid from 'material-ui/svg-icons/action/android';

import FontIcon from 'material-ui/FontIcon';
import { blueGrey50 } from 'material-ui/styles/colors';
import styles from './styles';

class AddressShow extends Component {
    constructor(props) {
        super(props);

        this.debug = false;
        if (this.debug) {
            console.log('ADDRESS-SHOW CONSTRUCTOR', this.props.address);
        }
    }

    log(message, message2 = '', message3 = '') {
        if (this.debug) {
            console.log(message, message2, message3);
        }
    }

    render() {
        if (this.debug) {
            console.log('ADDRESS-SHOW RENDER', this.props);
        }
        return (
            <div className="App">
                <RaisedButton
                    id={"leftTen"}
                    primary={true}
                    icon={
                        <FontIcon className="material-icons" color={blueGrey50}>
                            fast_rewind

                        </FontIcon>}
                    style={styles.button}
                    onClick={(e) => this.props.setAddress(-10, e)}
                />
                <RaisedButton
                    id={"leftOne"}
                    primary={true}
                    icon={
                        <FontIcon className="material-icons" color={blueGrey50}>
                            arrow_back_ios

                        </FontIcon>}
                    style={styles.button}
                    onClick={(e) => this.props.setAddress(-1, e)}
                />
                <RaisedButton
                    id={"fetchAddresses"}
                    label="Fetch Addresses"
                    labelPosition="before"
                    primary={true}
                    icon={
                        <FontIcon className="material-icons" color={blueGrey50}>
                            cloud_download
                        </FontIcon>
                    }
                    style={styles.button}
                    //onClick={this.props.setAddress}
                    // does is work to assign the address prop to the return of
                    // the fetchAddress call?  No--it says address is read-only.
                    //onClick={this.props.address=this.props.fetchAddress}
                    //onClick={(e) => {this.props.setAddress(1, e); this.disabled=true}}
                    onClick={(e) => this.props.setAddress(0, e)}

                />
                <RaisedButton
                    id={"rightOne"}
                    primary={true}
                    icon={
                        <FontIcon className="material-icons" color={blueGrey50}>
                            arrow_forward_ios

                    </FontIcon>}
                    style={styles.button}
                    onClick={(e) => this.props.setAddress(1, e)}
                />
                <RaisedButton
                    id={"rightTen"}
                    primary={true}
                    icon={
                        <FontIcon className="material-icons" color={blueGrey50}>
                            fast_forward

                        </FontIcon>}
                    style={styles.button}
                    onClick={(e) => this.props.setAddress(10, e)}
                />
                <p className="App-intro">
                    First Name: {this.props.address.firstName}
                </p>
                <p className="App-intro">
                    Last Name: {this.props.address.lastName}
                </p>
                <p className="App-intro">
                    Address: {this.props.address.address}
                </p>
                <p className="App-intro">City: {this.props.address.city}</p>
                <p className="App-intro">State: {this.props.address.state}</p>
                <p className="App-intro">Zip: {this.props.address.zip}</p>
                <p className="App-intro">Phone: {this.props.address.phone}</p>
                <p className="App-intro">Fax: {this.props.address.fax}</p>
                <p className="App-intro">
                    Toll Free: {this.props.address.tollfree}
                </p>
            </div>
        );
    }
}
AddressShow.propTypes = {
    address: PropTypes.shape({
        firstName: PropTypes.string,
        lastName: PropTypes.string,
        address: PropTypes.string,
        city: PropTypes.string,
        state: PropTypes.string,
        zip: PropTypes.string,
        phone: PropTypes.string,
        fax: PropTypes.string,
        tollfree: PropTypes.string
    }),
    setAddress: PropTypes.func,
    fetchAddress: PropTypes.func
};
export default AddressShow;
