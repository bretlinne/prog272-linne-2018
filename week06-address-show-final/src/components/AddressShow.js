import React, { Component } from 'react';
import '../App.css';
import '../address-list';
import PropTypes from 'prop-types';

class AddressShow extends Component {
    constructor(props) {
        super(props);

        this.debug = false;
        if(this.debug){

            console.log('ADDRESS-SHOW CONSTRUCTOR', this.props.address);
        }

    }

    log(message, message2 = '', message3 = '') {
        if (this.debug) {
            console.log(message, message2, message3);
        }
    }
    propTypes = {
        address: PropTypes.shape({
            firstName: PropTypes.string,
            lastName: PropTypes.string,
            address: PropTypes.string,
            city: PropTypes.string,
            state: PropTypes.string,
            zip: PropTypes.string,
            phone: PropTypes.string,
            fax: PropTypes.string,
            tollfree: PropTypes.string

        }),
        setAddress: PropTypes.func
    };
    render() {
        if(this.debug){ console.log('ADDRESS-SHOW RENDER', this.props);}
        return (
            <div className="App">
                <button id="showAddressClick" onClick={this.props.setAddress}>Show Address</button>

                <p className="App-intro">First Name: {this.props.address.firstName}</p>
                <p className="App-intro">Last Name: {this.props.address.lastName}</p>
                <p className="App-intro">Address: {this.props.address.address}</p>
                <p className="App-intro">City: {this.props.address.city}</p>
                <p className="App-intro">State: {this.props.address.state}</p>
                <p className="App-intro">Zip: {this.props.address.zip}</p>
                <p className="App-intro">Phone: {this.props.address.phone}</p>
                <p className="App-intro">Fax: {this.props.address.fax}</p>
                <p className="App-intro">Toll Free: {this.props.address.tollfree}</p>
            </div>
        );
    }
}

export default AddressShow;
