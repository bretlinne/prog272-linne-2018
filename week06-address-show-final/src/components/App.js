import React, { Component } from 'react';
import '../App.css';
import Address from './Address';
import Header from './Header';
import GetFile from './GetFile';


class App extends Component {
  render() {
    return (
      <div className="App">
          <Header></Header>
          <GetFile></GetFile>
          <Address></Address>
      </div>
    );
  }
}

export default App;
