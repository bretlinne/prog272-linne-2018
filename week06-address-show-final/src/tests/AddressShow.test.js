import React from 'react';
import ReactDOM from 'react-dom';
import App from '../components/App';
import AddressShow from '../components/AddressShow';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import AddressList from "../address-list";
configure({adapter: new Adapter()});


import addresses from '../address-list';

describe('AddressShow Shallow Suite', function() {
    const getIndex = (wrapper, index, talkToMe) => {
        if (debug || talkToMe) {
            const ninep = wrapper
                .find('div#addressShow')
                .childAt(index)
                .debug();
            console.log('NINEP:', ninep);
        }
    };

    let wrapper = null;

    const setAddress = () => {
        const address=addresses[1];
        wrapper.setProps({ address: address });
    };

    const addressTest = {
        firstName: 'Curly',
        lastName: 'Howard',
        address: '154 Wiseguy Ave',
        city: 'Washington',
        state: 'D.C.',
        zip: '20510',
        phone: '(555) 555-2621',
        fax: '(555) 555-0238',
        tollfree: '(555) 555-9186'
    };

    // this version has .dive at end of the const wrapper line.  This saves
    // some coding as in below.  We don't have to write .dive() calls where
    // they're used in the commented out version if we've used .dive in the wrapper
    // definition
    const defaultFieldTest = (name, index, talkToMe) => {
        const wrapper = shallow(<AddressShow address={addresses[0]} />).dive();
        console.log(wrapper.debug());
        const welcome = <p className="App-intro">{name}</p>;
        getIndex(wrapper, index, talkToMe);
        expect(wrapper.contains(welcome)).toEqual(true);
    };
    /*const defaultFieldTest = (name, index, talkToMe) => {
        const wrapper = shallow(<AddressShow address={addresses[0]} />);
        const welcome = <p className="App-intro">{name}</p>;
        getIndex(wrapper, index, talkToMe);
        expect(wrapper.dive().contains(welcome)).toEqual(true);
    };*/

    const afterClickFieldTest = (name) => {
        wrapper = shallow(<AddressShow
            address={addresses[0]}
            setAddress={setAddress}/>);
        const firstName = <p className="App-intro">{name}</p>;
        wrapper.dive().find('#setAddress').simulate('click');
        //console.log(wrapper.debug());
        expect(wrapper.dive().contains(firstName)).toBe(true);
    };

    fit('should take an AddressShow snapshot', () => {
        const elfTree = shallow(<AddressShow address={addresses[0]}/>);
        expect(elfTree).toMatchSnapshot();
    });

});

/*
describe('AddressShow Shallow Suite', function() {

    const debug = false;




    // * Display debug information about a DOM node of a component by index
    // * @param {object} wrapper - The HTML generated by your component
    // * @param {number} index - Index of HTML element you want to see.
    // * @param {boolean} talkToMe - Speak even if quiet is true

    const getIndex = (wrapper, index, talkToMe) => {
        if (debug || talkToMe) {
            const ninep = wrapper.find('div#addressShow').childAt(index).debug();
            console.log('NINEP:', ninep);
        }
    };



    const afterClickFieldTest = (name, index, talkToMe) => {
        const wrapper = shallow(<AddressShow address={AddressList[1]}/>);
        const welcome = <p className="App-intro">{name}</p>;
     //   console.log("being passed: ", name);
        getIndex(wrapper, index, talkToMe);
        expect(wrapper.contains(welcome)).toEqual(true);
    };

    it('renders and displays the first name', () => {
        defaultFieldTest('First Name: unknown', 0);
        afterClickFieldTest('First Name: ' + AddressList.firstName, 0);
    });

    it('renders and displays the last name', () => {
        defaultFieldTest('Last Name: unknown', 0);
        afterClickFieldTest('Last Name: ' + AddressList.lastName, 0);
    });

    it('renders and displays the address', () => {
        defaultFieldTest('Address: unknown', 0);
        afterClickFieldTest('Address: ' + AddressList.address, 0);
    });

    it('renders and displays the address', () => {
        //const addressTestValue = <p className="App-intro">'Address: unknown'</p>
        defaultFieldTest('Address: unknown', 0);
        afterClickFieldTest('Address: 154 Wiseguy Ave', 0);
    });

    it('renders and displays the city', () => {
        defaultFieldTest('City: unknown', 0);
        afterClickFieldTest('City: ' + AddressList.city, 0);
    });

    it('renders and displays the state', () => {
        defaultFieldTest('State: unknown', 0);
        afterClickFieldTest('State: ' + AddressList.state, 0);
    });

    it('renders and displays the zip', () => {
        defaultFieldTest('Zip: unknown', 0);
        afterClickFieldTest('Zip: ' + AddressList.zip, 0);
    });
    it('renders and displays the phone', () => {
        defaultFieldTest('Phone: unknown', 0);
        afterClickFieldTest('Phone: ' + AddressList.phone, 0);
    });
    it('renders and displays the fax', () => {
        defaultFieldTest('Fax: unknown', 0);
        afterClickFieldTest('Fax: ' + AddressList.fax, 0);
    });
    it('renders and displays the tollfree', () => {
        defaultFieldTest('Toll Free: unknown', 0);
        afterClickFieldTest('Toll Free: ' + AddressList.tollfree, 0);
    });
    // EVENTUALLY YOU WILL NEED TO WRITE MORE TESTS LIKE THE FIRST NAME TEST.
    // THE GOAL WILL BE TO TEST ALL THE PROPERTIES OF OUR COMPONENT.
    // AT FIRST, HOWEVER, JUST KEEP THESE TWO TESTS. WHEN THEY START
    // PASSING, THEN ADD TESTS FOR THE OTHER PROPERTIES SUCH AS LASTNAME...

});


*/