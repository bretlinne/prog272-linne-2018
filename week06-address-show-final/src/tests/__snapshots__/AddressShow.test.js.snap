// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressShow Shallow Suite should take an AddressShow snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddressShow
    address={
        Object {
            "address": "unknown",
            "city": "unknown",
            "fax": "unknown",
            "firstName": "unknown",
            "lastName": "unknown",
            "phone": "unknown",
            "state": "unknown",
            "tollfree": "unknown",
            "zip": "unknown",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          id="showAddressClick"
          onClick={undefined}
>
          Show Address
</button>,
        <p
          className="App-intro"
>
          First Name: 
          unknown
</p>,
        <p
          className="App-intro"
>
          Last Name: 
          unknown
</p>,
        <p
          className="App-intro"
>
          Address: 
          unknown
</p>,
        <p
          className="App-intro"
>
          City: 
          unknown
</p>,
        <p
          className="App-intro"
>
          State: 
          unknown
</p>,
        <p
          className="App-intro"
>
          Zip: 
          unknown
</p>,
        <p
          className="App-intro"
>
          Phone: 
          unknown
</p>,
        <p
          className="App-intro"
>
          Fax: 
          unknown
</p>,
        <p
          className="App-intro"
>
          Toll Free: 
          unknown
</p>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Show Address",
          "id": "showAddressClick",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Show Address",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "First Name: ",
            "unknown",
          ],
          "className": "App-intro",
        },
        "ref": null,
        "rendered": Array [
          "First Name: ",
          "unknown",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Last Name: ",
            "unknown",
          ],
          "className": "App-intro",
        },
        "ref": null,
        "rendered": Array [
          "Last Name: ",
          "unknown",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Address: ",
            "unknown",
          ],
          "className": "App-intro",
        },
        "ref": null,
        "rendered": Array [
          "Address: ",
          "unknown",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "City: ",
            "unknown",
          ],
          "className": "App-intro",
        },
        "ref": null,
        "rendered": Array [
          "City: ",
          "unknown",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "State: ",
            "unknown",
          ],
          "className": "App-intro",
        },
        "ref": null,
        "rendered": Array [
          "State: ",
          "unknown",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Zip: ",
            "unknown",
          ],
          "className": "App-intro",
        },
        "ref": null,
        "rendered": Array [
          "Zip: ",
          "unknown",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Phone: ",
            "unknown",
          ],
          "className": "App-intro",
        },
        "ref": null,
        "rendered": Array [
          "Phone: ",
          "unknown",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Fax: ",
            "unknown",
          ],
          "className": "App-intro",
        },
        "ref": null,
        "rendered": Array [
          "Fax: ",
          "unknown",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Toll Free: ",
            "unknown",
          ],
          "className": "App-intro",
        },
        "ref": null,
        "rendered": Array [
          "Toll Free: ",
          "unknown",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            id="showAddressClick"
            onClick={undefined}
>
            Show Address
</button>,
          <p
            className="App-intro"
>
            First Name: 
            unknown
</p>,
          <p
            className="App-intro"
>
            Last Name: 
            unknown
</p>,
          <p
            className="App-intro"
>
            Address: 
            unknown
</p>,
          <p
            className="App-intro"
>
            City: 
            unknown
</p>,
          <p
            className="App-intro"
>
            State: 
            unknown
</p>,
          <p
            className="App-intro"
>
            Zip: 
            unknown
</p>,
          <p
            className="App-intro"
>
            Phone: 
            unknown
</p>,
          <p
            className="App-intro"
>
            Fax: 
            unknown
</p>,
          <p
            className="App-intro"
>
            Toll Free: 
            unknown
</p>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Show Address",
            "id": "showAddressClick",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Show Address",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "First Name: ",
              "unknown",
            ],
            "className": "App-intro",
          },
          "ref": null,
          "rendered": Array [
            "First Name: ",
            "unknown",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Last Name: ",
              "unknown",
            ],
            "className": "App-intro",
          },
          "ref": null,
          "rendered": Array [
            "Last Name: ",
            "unknown",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Address: ",
              "unknown",
            ],
            "className": "App-intro",
          },
          "ref": null,
          "rendered": Array [
            "Address: ",
            "unknown",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "City: ",
              "unknown",
            ],
            "className": "App-intro",
          },
          "ref": null,
          "rendered": Array [
            "City: ",
            "unknown",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "State: ",
              "unknown",
            ],
            "className": "App-intro",
          },
          "ref": null,
          "rendered": Array [
            "State: ",
            "unknown",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Zip: ",
              "unknown",
            ],
            "className": "App-intro",
          },
          "ref": null,
          "rendered": Array [
            "Zip: ",
            "unknown",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Phone: ",
              "unknown",
            ],
            "className": "App-intro",
          },
          "ref": null,
          "rendered": Array [
            "Phone: ",
            "unknown",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Fax: ",
              "unknown",
            ],
            "className": "App-intro",
          },
          "ref": null,
          "rendered": Array [
            "Fax: ",
            "unknown",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Toll Free: ",
              "unknown",
            ],
            "className": "App-intro",
          },
          "ref": null,
          "rendered": Array [
            "Toll Free: ",
            "unknown",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
